src/Plugin/WebformElement/CvrLookupElement.php

```diff
@@ -0,0 +1,152 @@
+    LoggerInterface $logger,
+    ConfigFactoryInterface $config_factory,
+    AccountInterface $current_user,
+    EntityTypeManagerInterface $entity_type_manager,
+    ElementInfoManagerInterface $element_info,
+    WebformElementManagerInterface $element_manager,
+    WebformTokenManagerInterface $token_manager,
+    WebformLibrariesManagerInterface $libraries_manager,
+    AuthProviderService $authProviderService,
+    CvrServiceInterface $cvrService
+  ) {
+    // PluginBase::__construct() accepts only three arguments.
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->configFactory = $config_factory;
+    $this->currentUser = $current_user;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->elementInfo = $element_info;
+    $this->elementManager = $element_manager;
+    $this->tokenManager = $token_manager;
+    $this->librariesManager = $libraries_manager;
+    $this->authProviderService = $authProviderService;
+    $this->cvrService = $cvrService;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory')->get('webform'),
+      $container->get('config.factory'),
+      $container->get('current_user'),
+      $container->get('entity_type.manager'),
+      $container->get('plugin.manager.element_info'),
+      $container->get('plugin.manager.webform.element'),
+      $container->get('webform.token_manager'),
+      $container->get('webform.libraries_manager'),
+      $container->get('os2web_nemlogin.auth_provider'),
+      $container->get('os2forms_cvr_lookup.service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function handleElementPrepopulate(array &$element, FormStateInterface &$form_state) {
+    $prepopulateKey = $this->getPrepopulateFieldFieldKey();
+    // Fetch value from cvr lookup.
+    $data = NULL;
+
+    if ($form_state->has(static::FORM_STATE_DATA)) {
+      $data = $form_state->get(static::FORM_STATE_DATA);
+    }
+    else {
+      // Making the request to the plugin, and storing the information on the
+      // form, so that it's available on the next element within the same
+      // webform render.
+      $plugin = $this->authProviderService->getActivePlugin();
+
+      if ($plugin->isAuthenticated()) {
+        try {
+          $cvr = $plugin->fetchValue("cvr");
+          if ($cvr) {
+            $result = $this->cvrService->search($cvr);
+            $data = $result->toArray();
+            // Add data for the CVR value element.
+            $data['cvr'] = $cvr;
+            // Merge in some values from the NemID login provider.
+            $data += array_filter(
+              array_map(
+                [$plugin, 'fetchValue'],
+                [
+                  'pid' => 'pid',
```

Comment:
Does CVR data contain a “pid” (person(al) id) key?

Most programming is just stealing code (sometimes from oneself), but sometimes the loot needs a little clean-up.